### DISCUSSION TITLE
Bootstrap 4 [ col ou css grid layout ]

### DISCUSSION THREAD

**Discussion by kadumx** [frontendbr/forum#1194] [Perguntas e Respostas]
Olá pessoal,
estou desenvolvendo um projeto usando Bootstrap 4. 

Na hora de fazer o grid do conteúdo, cheguei a dúvida usar col ou css grid layout. 

Sabendo que o Col e "nativo" do bootstrap e o Css Grid Layout não, qual seria a abordagem de vocês e a ideal usando Bootstrap 4!?

Obrigado

**Comment by luizbills** [comment]
Use o que você mais tem familiaridade.
Alias não acredito que bootstrap vai ter classes para Grid Layout, pois grid não fica legal como um framework. Grid layout depende muito do projeto e uma solução generica vai mais atrapalhar do que ajudar.

**Comment by diegodsgarcia** [comment]
Concordo com o Luiz a cima! E também só queria acrescentar que o Grid do Bootstrap foi justamente criado para atender as necessidades que o CSS nativo não tinha suporte em épocas onde só existia a propriedade Float. Nos dias de hoje não faz mais sentido o Framework criar classes que nativamente o próprio CSS já faz (como o display GRID ou FLEX faz).

**Comment by kadumx** [comment]
Obrigado, @luizbills e @diegodsgarcia

**Comment by evertonthepaula** [comment]
Qual você utilizaria se não tivesse o Bootstrap? Pense que seu trabalho é cortar madeira, você pode usar uma serra de mão ou um facão, este é o CSS puro. Agora você pode usar o Bootstrap como uma ferramenta para trabalhar com coisa maiores, substituindo o facão pelo machado e a serra de mão pela serra elétrica. No fim é assim que eu vejo, tudo é ferramenta e você usa a ferramenta certa para o trabalho certo, e se der você usa uma ferramenta maior ou mais automatizada para fazer a mesma coisa.

**Comment by guilhermecz9** [comment]
Será que a dupla Bootstrap/jQuery irá sobreviver à evolução do EcmaScript/CSS e a forte adoção dos Browsers?

Todos os dias a comunidade entrega soluções, que antes pareciam impossíveis de se alcançar sem um framework, utilizando abordagens simples e sofisticadas de puro javascript e css, como é o caso do Grid/Flexbox.

Às vezes eu penso que o futuro dos frameworks não é lá muito bom. O próprio Angular pode vir a ser superado no curto prazo, e isso vale para Vue, React, Lodash, etc.

Tenho enfrentado outra questão: Gasto algum tempo investindo no Laravel para daqui a algum tempo ele ser superado por outro framework ou pelo próprio PHP, ou fico com o PHP7 puro?

A questão é a mesma: usar ou não um Framework.

Tenho feito esse questionamento com relação ao Bootstrap/jQuery, mas em princípio não vejo problemas em incorporar flexbox nativo e display grid em meio ao HTML com Bootstrap, que por mais organizado e bem feito que seja, é também uma "colcha de retalhos".

**Comment by felipefialho** [comment]
@guilhermecruzdev Sugiro abrir uma nova issue para debater isso 😄

**Comment by evertonthepaula** [comment]
@guilhermecruzdev respondo sua pergunta com a mesma resposta, acrescentando que a especificação oficial anda sempre mais lenta que a comunidade, quando por exemplo a W3C ainda estuda Web Components nós já utilizamos eles amplamente nas nossas SPA's, utilizando as especificações de como eles irão funcionar um dia, graças aos Frameworks, as Bibliotecas e aos Polyfills, ou poderia dizer, graças a estas ferramentas :p . Para o Laravel acredito que seja a mesma coisa, o que o Laravel faz para automatizar seu trabalho é o que você faria na mão se não tivesse o Laravel? A arquitetura que ele te disponibiliza é uma que você concorda que funciona? As regras de segurança que ele implementa são eficientes? Tem várias perguntas que você pode fazer. Não sou especialista no assunto, mas está muito claro para mim que tudo é ferramenta e se a ferramenta faz exatamente o que você faria na mão, use ela, pois será mais rápido, organizado e seguro :) .

**Comment by guilhermecz9** [comment]
@evertonthepaula Obrigado pela resposta. Faltou um ponto crucial, que é o da curva de aprendizado. Eu tenho um projeto pra entregar em 1 mês, fico com o dilema: gasto os primeiros 20 dias para estudar e aprender a dominar o Laravel, ou React ou Vue, (estamos falando aqui de curva de aprendizado), investindo agora para "ganhar" ali na frente, ou já começo produzindo com o bom e velho PHP 7.x que tanto domino? Questão difícil.

Não é nada bom desejar que um dia os frameworks deixem de existir ou sejam reduzidos, até porque são eles que muitas vezes inovam e ditam a tendência do mercado, as linguagens copiam as estruturas dos frameworks e das outras linguagens, incorporando "o que deu certo".

**Comment by ViniciusGularte** [comment]
@guilhermecruzdev  uma boa issue para se debater.
